code/game/objects/items/weapons/holy_weapons.dm

```diff
@@ -445,5 +445,130 @@
+		return 0
+	if(!ishuman(user))		//prevents the horror (runtimes) of missionary xenos and other non-human mobs that might be able to activate the item
+		return 0
+	var/mob/living/carbon/human/missionary = user
+	if(missionary.wear_suit && istype(missionary.wear_suit, /obj/item/clothing/suit/hooded/chaplain_hoodie/missionary_robe))
+		var/obj/item/clothing/suit/hooded/chaplain_hoodie/missionary_robe/robe_to_link = missionary.wear_suit
+		if(robe_to_link.linked_staff)
+			to_chat(missionary, "<span class='warning'>These robes are already linked with a staff and cannot support another. Connection refused.</span>")
+			return 0
+		robes = robe_to_link
+		robes.linked_staff = src
+		to_chat(missionary, "<span class='notice'>Link established. Faith generators initialized. Go spread the word.</span>")
+		faith = 100		//full charge when a fresh link is made (can't be delinked without destroying the robes so this shouldn't be an exploitable thing)
+		return 1
+	else
+		to_chat(missionary, "<span class='warning'>You must be wearing the missionary robes you wish to link with this staff.</span>")
+		return 0
+
+/obj/item/weapon/nullrod/missionary_staff/afterattack(mob/living/carbon/human/target, mob/living/carbon/human/missionary, flag, params)
+	if(!istype(target) || !istype(missionary))		//ishuman checks effectively
+		return
+	if(target == missionary)	//you can't convert yourself, that would raise too many questions about your own dedication to the cause
+		return
+	if(!robes)		//staff must be linked to convert
+		to_chat(missionary, "<span class='warning'>You must link your staff to a set of missionary robes before attempting conversions.</span>")
+		return
+	if(!missionary.wear_suit || missionary.wear_suit != robes)	//must be wearing the robes to convert
+		return
+	if(faith < 100)
+		to_chat(missionary, "<span class='warning'>You don't have enough faith to attempt a conversion right now.</span>")
+		return
+	to_chat(missionary, "<span class='notice'>You concentrate on [target] and begin the conversion ritual...</span>")
+	if(!target.mind)	//no mind means no conversion, but also means no faith lost.
+		to_chat(missionary, "<span class='warning'>You halt the conversion as you realize [target] is mindless! Best to save your faith for someone more worthwhile.</span>")
+		return
+	if(do_after(missionary, 40))	//4 seconds to temporarily convert, roughly 1 second faster than a vampire's enthrall ability
+		if(faith < 100)		//to stop people from trying to exploit the do_after system to multi-convert, we check again if you have enough faith when it completes
+			to_chat(missionary, "<span class='warning'>You don't have enough faith to complete the conversion on [target]!</span>")
+			return
+		if(missionary in viewers(target))	//missionary must maintain line of sight to target, but the target doesn't necessary need to be able to see the missionary
+			do_convert(target, missionary)
+			return
```

Comment:
ok, you got me on these, i was a bit over zealous on pasting that response on the trailing return bits and mistook this (and the next one) for non-trailing ones in the other proc. my bad